app/models/spree/address.rb

```diff
@@ -10,12 +10,16 @@ class Address < ActiveRecord::Base
     has_one :enterprise, dependent: :restrict_with_exception
     has_many :shipments
 
+    before_validation :geocode, if: :use_geocoder?
```

Comment:
Why are you moving the geocoding before the validation? That means that we may geocode even if the enterprise isn't valid which could increase our API usage and slow things down.

Reply:
Good point @mkllnk I have fixed this now in dc88d7d, thanks